solr/core/src/java/org/apache/solr/search/FunctionQParser.java

```diff
@@ -361,7 +361,9 @@ protected ValueSource parseValueSource(int flags) throws SyntaxError {
         ((FunctionQParser)subParser).setParseMultipleSources(true);
       }
       Query subQuery = subParser.getQuery();
-      if (subQuery instanceof FunctionQuery) {
+      if (subQuery == null) {
+        valueSource = new DoubleConstValueSource(0.0f);
```

Comment:
Why a Double?

Reply:
Having a second look I see that a ConstValueSource is a better alternative, I'll update this.